java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator;
+
+import com.github.javaparser.ast.CompilationUnit;
+import io.fabric8.java.generator.exceptions.JavaGeneratorException;
+import io.fabric8.java.generator.nodes.AbstractJSONSchema2Pojo;
+import io.fabric8.java.generator.nodes.GeneratorResult;
+import io.fabric8.java.generator.nodes.JCRObject;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionSpec;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionVersion;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.io.File;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class CRGeneratorRunner {
+
+    public void run(File source, File basePath) {
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
+            // Parse CRD with fabric8
+            CustomResourceDefinition crd =
+                    client.apiextensions().v1().customResourceDefinitions().load(source).get();
```

Comment:
Why not just use the Serialization utility class here, no need to create a client instance

Reply:
thanks 👍  yes, using `Serialization` now